src/dogecoin.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015 The Dogecoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <boost/random/uniform_int.hpp>
+#include <boost/random/mersenne_twister.hpp>
+
+#include "dogecoin.h"
+
+int static generateMTRandom(unsigned int s, int range)
+{
+    boost::mt19937 gen(s);
+    boost::uniform_int<> dist(1, range);
+    return dist(gen);
+}
+
+CAmount GetDogecoinBlockSubsidy(int nHeight, const Consensus::Params& consensusParams, uint256 prevHash)
+{
+    int halvings = nHeight / consensusParams.nSubsidyHalvingInterval;
+
+    if (nHeight < 145000) // && !consensusParams.SimplifiedRewards())
```

Comment:
Why is this commented out? Regtest used to generate static rewards which makes the tests cleaner.